source/core/constants.ts

```diff
@@ -9,15 +9,19 @@ export const supportsRequestStreams = (() => {
 	const supportsRequest = typeof globalThis.Request === 'function';
 
 	if (supportsReadableStream && supportsRequest) {
-		hasContentType = new globalThis.Request('https://empty.invalid', {
-			body: new globalThis.ReadableStream(),
-			method: 'POST',
-			// @ts-expect-error - Types are outdated.
-			get duplex() {
-				duplexAccessed = true;
-				return 'half';
-			},
-		}).headers.has('Content-Type');
+		try {
+			hasContentType = new globalThis.Request('https://empty.invalid', {
+				body: new globalThis.ReadableStream(),
+				method: 'POST',
+				// @ts-expect-error - Types are outdated.
+				get duplex() {
+					duplexAccessed = true;
+					return 'half';
+				},
+			}).headers.has('Content-Type');
+		} catch {
+			// IOS QQBrowser throw "unsupported BodyInit type" Error (see issue #581)
```

Comment:
Sorry, but we cannot silently ignore all errors from the `Request` constructor. There are many genuine, useful errors that it can potentially throw. Maybe you could explain what you are trying to accomplish (I couldn't understand from the linked issue). If you think we should be ignoring a particular error in certain circumstances, then we will need to properly detect those conditions and the specific error.

Reply:
Based on my understanding, this code is supposed to determine whether the runtime supports streaming requests. However, the [code](https://github.com/edram/issue-ky-581/blob/main/src/main.ts#L14) above throws `Error: unsupported BodyInit type` in the QQ app or QQ browser on iOS. This indicates that these two runtimes do not support streaming requests. The check for support should not throw an error. Given the large number of QQ users, the latest version of ky in my project can use is version 0.31.1.

![image](https://github.com/sindresorhus/ky/assets/14024542/b42ab36c-f7b9-4d8c-b620-4d5c8a679853)

Reply:
Okay, thanks. I understand now. I have reopened the PR. Can you please update your code so it re-throws the error if `error.message !== 'unsupported BodyInit type'`?

Reply:
Thank you for your response. I have updated my code. 🫰